### PR TITLE
Fix false positive tests

### PR DESCRIPTION
**Pull request by variadico**
Currently, the makefile ignores Go test errors because it wants to
always run clean up. This change refactors the makefile to allow Go test
errors to be noticed and ensure that clean up runs.

In addition, it enables GODEBUG=x509ignoreCN=0 during tests to give us
some time to figure out how to generate certs with SANs. This is
temporary and should be removed soon.

### CODE REVIEW

**Comment by variadico** [comment]
Also, @ColinSullivan1 brought up about users not using SANs. https://github.com/nats-io/nats-kafka/pull/22#discussion_r557040619

Should our CI only test for Go 1.14? We can temporarily revert the deprecated behavior in Go 1.15 with`GODEBUG`. If we release nats-kafka under Go 1.15, users will have to set `GODEBUG` in their environment if they don't want to use SANs.

**Review by wallyqs — APPROVED** [review]
LGTM

**Comment by wallyqs** [comment]
GODEBUG for testing is fine, though I think that maybe for the next release we still use Go 1.14 (the impact of Go 1.15 is that when nats-kafka tries to connect to a server then it will be more strict with the certs that are used).
You can find how these were generated  for the nats-server here: https://github.com/nats-io/nats-server/blob/a9a6bdc04fc3283b2aeef37e643ba3db52069e89/test/tls_test.go#L1368-L1375

**Comment by ColinSullivan1** [comment]
1.15 for testing is OK imo, but release should be 1.14 imo.  I think release binaries built from 1.15 would create issues for too many users due to the SAN requirement.

File: Makefile [hunk 1]
```diff
@@ -1,50 +1,67 @@
-build: fmt check compile
+# FIXME: Need to figure out how to generate certs with SANs.
```
Comment (ColinSullivan1) [inline Makefile:1]: Old comment?

File: docs/buildandrun.md [hunk 2]
```diff
@@ -2,29 +2,37 @@
 
 ## Running the server
 
-The server will compile to an executable named `nats-kafka`. A [configuration](config.md) file is required to get any real behavior out of the server.
+The server will compile to an executable named `nats-kafka`. A
+[configuration](config.md) file is required to get any real behavior out of the
```
Comment (ColinSullivan1) [inline docs/buildandrun.md:6]: What happens w/o a config?  If it's a NOOP, then probably should require one.

File: server/core/nats2kafka_test.go [hunk 3]
```diff
@@ -42,13 +42,13 @@ func TestSimpleSendOnNatsReceiveOnKafka(t *testing.T) {
...
-	require.NoError(t, err)
-
 	reader := tbs.CreateReader(topic, 5000)
 	defer reader.Close()
 
-	_, data, err := tbs.GetMessageFromKafka(reader, 5000)
+	err = tbs.NC.Publish("test", []byte(msg))
+	require.NoError(t, err)
+
+	_, data, err := tbs.GetMessageFromKafka(reader, 10000)
```
Comment (ColinSullivan1) [inline server/core/nats2kafka_test.go:51]: ...could create a constant for the timeout duration.

**Review by ColinSullivan1 — APPROVED** [review]
One question (can be address outside of this PR) and a nitpick - take it or leave it.  LGTM!

File: server/core/nats2kafka_test.go (hunk 3 above)
Comment (variadico) [reply server/core/nats2kafka_test.go:51]: Hm. Yeah, thinking about what to do here. Changing the timeouts actually didn't help, so I'll probably revert this change. I'm wondering if there are some more fundamental tweaks we need. The tests failing on CI also fail on my laptop, if I run the tests enough times. CI is really good at exposing them though! :smile:

**Comment by variadico** [comment]
Okay, so we have some flaky tests. CI is really good at exposing them. When I run them enough times, they fail locally on my Linux laptop as well.

For now, I've disabled the flaky tests in `make test`, which is what CI uses. However, `make test-failfast` will run everything, including flakes, and will stop running after the first failure. Overall, we'll actually be alerted of failures now, so that's still good.

I'm gonna handle the config issue in a different PR.

Also, as for fixing the flakes, I'm also working on a PR to swap out segmentio/kafka-go for Shopify/sarama because the former has some bugs that cause unwanted batching. I'm hoping that will stabilize the flakes, otherwise I'll take another look.

**Pull request merged**